server/src/main/java/io/crate/expression/scalar/arithmetic/ExpFunction.java

```diff
@@ -21,31 +21,28 @@
 
 package io.crate.expression.scalar.arithmetic;
 
+import static io.crate.metadata.functions.Signature.scalar;
+
 import io.crate.expression.scalar.ScalarFunctionModule;
 import io.crate.expression.scalar.UnaryScalar;
 import io.crate.metadata.Scalar;
 import io.crate.types.DataTypes;
 
-import static io.crate.metadata.functions.Signature.scalar;
-
 public class ExpFunction {
 
     public static final String NAME = "exp";
 
     public static void register(ScalarFunctionModule module) {
-        for (var type : DataTypes.NUMERIC_PRIMITIVE_TYPES) {
-            var typeSignature = type.getTypeSignature();
-            module.register(
-                scalar(NAME, typeSignature, typeSignature)
-                    .withFeature(Scalar.Feature.NULLABLE),
-                (declaredSignature, boundSignature) ->
-                    new UnaryScalar<>(
-                        declaredSignature,
-                        boundSignature,
-                        type,
-                        x -> type.sanitizeValue(Math.exp(((Number) x).doubleValue()))
```

Comment:
I guess we don't need the `sanitizeValue` anymore because the argument is already converted to double by the function signature resolution mechanism, right?

Reply:
```
    public final R evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<T>... args) {
        assert args.length == 1 : "UnaryScalar expects exactly 1 argument, got: " + args.length;
        T value = type.sanitizeValue(args[0].value());
        if (value == null) {
            return null;
        }
        return func.apply(value);
    }
```
It is already sanitized, this could be its own clean up commit actually. Thanks for pointing out.

Reply:
ah, yes in `UnaryScalar`, didn't think to check it there, thx!

Reply:
Sorry for confusion @matriv, we needed to keep the `sanitize` call actually. It was sanitizing/casting the _output_ to the target type.